src/guide/deployment/lamp.rst

```diff
@@ -0,0 +1,160 @@
+==========================
+How to set up a LAMP stack
+==========================
+
+Context: fc platform, user management, multi vm?
+
+
+The Flying Circus Platform provides components which can be used to set up an environment for operating PHP applications. The configuration *can* be automated with deployment tools of course, but this is not part of this tutorial.
+
+What you should know before:
+
+* The general ideas of the Flying Circus Platform: :ref:`firststeps`.
+* The difference between human users and service users: :ref:`useraccounts`
+* How to connect to VMs via SSH (or SFTP): :ref:`connecting`
+
+Assumptions:
+
+* There is only *one* VM, ``myvm00``. The setup generally works in multi VM setups, too, though.
+* You will host multiple sites in one setup.
+* Each site is being hosted in a different service user. We use ``s-myserviceuser`` here. Switch to the service user with ``sudo -iu s-myserviceuser``. The docroot is *inside* the service user's home directory.
+
+
+General structure
+=================
+
+The general request flow for our LAMP stack is as follows:
+
+* NGINX: SSL termination
+* Apache: Used so ``.htaccess`` files work.
+* PHP-FPM for actually running PHP code
+
+In addition you likely want a variant of MySQL.
+
+
+Component setup
+===============
+
+MySQL
+*****
+
+Enable a MySQL role of your choice, e.g. mysql57, percona80, in the VM dashboard on https://my.flyingcircus.io/. See :ref:`nixos-mysql` for details.
+
+If you just enabled a MySQL role it might now be available on the VM, yet. Call ``sudo fc-manage -eb`` to force a build. The ``-e`` option is there to reload the configuration you have set in the VM dashboard, so it's usually omitted.
+
+To create a service specific MySQL user and database you switch to the mysql user and start a mysql shell:
+
+.. code-block:: Shell
+
+    $ sudo -iu mysql
+    mysql$ mysql
+
+    mysql> CREATE DATABASE myservice;
+    mysql> CREATE USER myservice@'%' IDENTIFIED BY 'a password';
+    mysql> GRANT ALL on myservice.* TO myservice@'%';
+
+
+You will of course need username and password for your application. We will not go into detail here.
+
+Note that we used ``myservice@'%'`` as user name. This generally allows access from everywhere (``%``). Our firewall rules make sure that MySQL is only reachable via the SRV interface and only from other VMs in the *same* resource group. See :ref:`firewall` for details.
+
+
+Apache / FPM
+************
+
+You configure Apache and FPM in one step. With the Flying Circus ``lamp`` role you define an Apache docroot which is properly hooked up with FPM.
+
+But first, there are some generic Apache settings, which are used for *all* sites. :
+
+.. code-block:: Nix
+
+    # /etc/local/nixos/apache.nix
+    { pkgs, ... }:
+
+    {
+
+      flyingcircus.roles.lamp = {
+
+        # Configure PHP version:
+        php = pkgs.lamp_php74;
+
+        # Additional Apache configuration
+        apache_conf = ''
+          MaxRequestWorkers 5
```

Comment:
I'd leave this one out in the tutorial. Otherwise people start always copying this, overriding the quite sensible default.